scanner_cli.py

```diff
@@ -63,40 +69,93 @@ def fetch_properties(state_code):
-
-            # Extract relevant information
-            for single_property in property_list:
-                property_data = [(single_property["property_id"],
-                                single_property["location"]["address"]["line"],
-                                single_property["location"]["address"]["city"],
-                                single_property["location"]["address"]["state_code"],
-                                single_property["location"]["address"]["postal_code"],
-                                single_property["list_price"],
-                                single_property["description"]["beds"],
-                                single_property["description"]["baths"],
-                                single_property["primary_photo"]["href"])]
-                writer.writerows(property_data)
+        file_name = "real_estate_for_sale_" + state_code + ".json"
+
+        # Extract specific information from each property
+        filtered_properties = [
+            {
+                "property_id": property["property_id"],
+                "address": {
+                    "line": property["location"]["address"]["line"],
+                    "city": property["location"]["address"]["city"],
+                    "state_code": property["location"]["address"]["state_code"],
+                    "postal_code": property["location"]["address"]["postal_code"]
+                },
+                "list_price": property["list_price"],
+                "description": {
+                    "beds": property["description"]["beds"],
+                    "baths": property["description"]["baths"]
+                },
+                "primary_photo": property["primary_photo"]
+            }
+            for property in property_list
+        ]
+
+        # Write to JSON
+        with open(file_name, mode='w', encoding="utf-8") as file:
+            json.dump(filtered_properties, file, indent=2)
 
         print(f"Retrieved newest 42 properties for sale in {state_code}")
         print(f"Saved to {file_name}")
     except requests.exceptions.RequestException as ex:
         print(f"Error: {ex}")
 
+def store_user_data(username):
+    """
+    Store and calculate metrics based on user data.
+
+    Parameters:
+    - username (str): Username of the user, used as a reference for stored values.
+
+    Returns:
+    None: Updates the json file user_data.json
+    """
+    # Load existing data from file
+    try:
+        with open('user_data.json', 'r', encoding="utf8") as file:
+            user_data = json.load(file)
+        # Check if the username is already in the database
+        if username in user_data:
+            print(f"Hello, {username}!")
+        else:
+            print(f"New user {username}")
+    except FileNotFoundError:
+        # Create a new user_data dictionary if the file doesn't exist
+        user_data = {}
+        print(f"New user {username}")
+
+    # Ask for and save user input
+    user_data[username] = {
```

Comment:
if username already exists do we want to enter in all of this information again?

Reply:
The reason I made it so that we would re-enter this info again was so that we have the option to update the fields. I think what I will do is output the profile if it exists and ask if they wanted to update the fields.